[BUG] updating a cloudfoundry_app routinely crashes the provider

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What version of the Terraform provider are you using?

1.1.0

### What version of the Terraform CLI are you using?

1.9.8

### What CF API version are you using?

3.182.0

### What type of issue are you facing

bug report

### Describe the bug

Running `terraform apply` to re-deploy an app is routinely crashing for several simple `ruby_buildpack` apps. I think the deploy is actually working (most of the time, anyway) but the terraform state is not properly updated afterwards.

### Expected Behavior

Terraform does not crash

### Steps To Reproduce

_No response_

### Your Terraform Configuration

_No response_

### Roles

_No response_

### Add screenshots to help explain your problem

_No response_

### Additional context

console output:

```
cloudfoundry_app.app: Still modifying... [id=c7725c79-f5b9-420d-82bc-034247e5259f, 1m40s elapsed]
╷
│ Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-cloudfoundry_v1.1.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105017a34]

goroutine 73 [running]:
github.com/cloudfoundry/go-cfclient/v3/operation.(*AppPushOperation).pushRollingApp(0x140003f0518, {0x1057d0918, 0x14000548cf0}, 0x140003fe7e0, 0x14000199b80, {0x1057c6b60, 0x14000434628})
        github.com/cloudfoundry/go-cfclient/v3@v3.0.0-alpha.9/operation/push.go:158 +0x234
github.com/cloudfoundry/go-cfclient/v3/operation.(*AppPushOperation).pushWithStrategyApp(0x140003f0518?, {0x1057d0918?, 0x14000548cf0?}, 0x1400014e3c0?, 0x24?, {0x1057c6b60?, 0x14000434628?})
        github.com/cloudfoundry/go-cfclient/v3@v3.0.0-alpha.9/operation/push.go:68 +0x40
github.com/cloudfoundry/go-cfclient/v3/operation.(*AppPushOperation).Push(0x140003f0518, {0x1057d0918, 0x14000548cf0}, 0x14000199b80, {0x1057c6b60, 0x14000434628})
        github.com/cloudfoundry/go-cfclient/v3@v3.0.0-alpha.9/operation/push.go:61 +0x84
github.com/cloudfoundry/terraform-provider-cloudfoundry/internal/provider.(*appResource).push(_, {{0x2, {0x14000038c60, 0x1b}}, {0x2, {0x14000408bc0, 0xd}}, {0x2, {0x14000038ce0, 0x1c}}, ...}, ...)
        github.com/cloudfoundry/terraform-provider-cloudfoundry/internal/provider/resource_app.go:493 +0x18c
github.com/cloudfoundry/terraform-provider-cloudfoundry/internal/provider.(*appResource).upsert(0x1400007e528, {0x1057d0918, 0x14000548cf0}, 0x1400005b118, 0x140003f1148, 0x140005b5310, 0x140005b5348)
        github.com/cloudfoundry/terraform-provider-cloudfoundry/internal/provider/resource_app.go:452 +0x504
github.com/cloudfoundry/terraform-provider-cloudfoundry/internal/provider.(*appResource).Update(0x1400007e908?, {0x1057d0918?, 0x14000548cf0?}, {{{{0x1057d6a18, 0x140006ed830}, {0x1056785a0, 0x140000bdc80}}, {0x1057d9f60, 0x14000320960}}, {{{0x1057d6a18, ...}, ...}, ...}, ...}, ...)
        github.com/cloudfoundry/terraform-provider-cloudfoundry/internal/provider/resource_app.go:426 +0x38
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).UpdateResource(0x140001878c8, {0x1057d0918, 0x14000548cf0}, 0x140003f1458, 0x140003f1430)
        github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_updateresource.go:122 +0x53c
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x140001878c8, {0x1057d0918, 0x14000548cf0}, 0x140007044b0, 0x140003f1620)
        github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_applyresourcechange.go:102 +0x118
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x140001878c8, {0x1057d0918?, 0x14000548c00?}, 0x14000704410)
        github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/proto6server/server_applyresourcechange.go:55 +0x2e0
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x1400020bcc0, {0x1057d0918?, 0x140005481e0?}, 0x14000344310)
        github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov6/tf6server/server.go:865 +0x2a8
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1057a0580, 0x1400020bcc0}, {0x1057d0918, 0x140005481e0}, 0x14000256080, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:611 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001a9200, {0x1057d0918, 0x14000548150}, {0x1057d8000, 0x14000376000}, 0x1400069a000, 0x140002e6ed0, 0x105df2958, 0x0)
        google.golang.org/grpc@v1.67.1/server.go:1394 +0xb64
google.golang.org/grpc.(*Server).handleStream(0x140001a9200, {0x1057d8000, 0x14000376000}, 0x1400069a000)
        google.golang.org/grpc@v1.67.1/server.go:1805 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.67.1/server.go:1029 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 20
        google.golang.org/grpc@v1.67.1/server.go:1040 +0x13c

Error: The terraform-provider-cloudfoundry_v1.1.0 plugin crashed!
```

--- a/internal/provider/resource_app.go
+++ b/internal/provider/resource_app.go
@@ -445,6 +445,10 @@ func (r *appResource) upsert(ctx context.Context, reqPlan *tfsdk.Plan, reqState 
 	if respDiags.HasError() {
 		return
 	}
+	//Required to set instances value for rolling strategy app update for go-cfclient. So setting it to cf-default of 1
+	if appManifestValue.Instances == nil {
+		appManifestValue.Instances = uinttouintptr(1)
+	}
 	if reqState != nil {
 		diags = reqState.Get(ctx, &previousState)
 		respDiags.Append(diags...)

